UniversityApp/Pagination/PagedModel.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace UniversityApp.Pagination
+{
+    public class PagedModel<T>
+    {
+        const int maxPageSize = 50;
+
+        public int PageSize { get; set; }
+        public int PageNo { get; set; }
+        public int TotalRecordCount { get; set; }
+        public int PageCount { get; set; }
+        public List<T> Data { get; set; } = new List<T>();
+
+        public PagedModel(List<T> data, int pageNo, int pageSize, int totalRecordCount)
+        {
+            Data = data;
+            PageNo = pageNo;
+            PageSize = (pageSize > maxPageSize) ? maxPageSize : pageSize;
```

Comment:
I wouldn't set requested pageSize to default as client calling search endpoint expects  data according requested parameters and gets completely different data without any information. Instead setting those values during executing request I would add validation. If pageSize is bigger than allowed then inform client and return BadRequest(). This way client will get specific information how this endpoint goes

Reply:
with PageCount - if client requested page that "not exists" then empty data returned

Reply:
PageCount - to be deleted